exchanges/stream/websocket.go

```diff
@@ -205,14 +206,6 @@ func (w *Websocket) Connect() error {
 		w.connectionMonitor()
 	}
 
-	// Resubscribe after re-connection
-	if len(w.subscriptions) != 0 {
-		err = w.Subscriber(w.subscriptions)
-		if err != nil {
-			return fmt.Errorf("%v Error subscribing %s", w.exchangeName, err)
-		}
-	}
-
```

Comment:
The removal of this is the reason why we're not resubscribing after a disconnection. Upon reconnection, we get to [this](https://github.com/thrasher-corp/gocryptotrader/pull/650/files#diff-81b63f155e3eb371602734d39949da22f7752344a95cf380308c9376517ca1cbR470) line and `w.subscriptions` is already populated. So then the difference in subscriptions from the provided to `w.subscriptions` is zero, so no resubscribe occurs.
Either add this back to allow for subscriptions to reoccur on disconnection or set w.subscriptions to nil upon disconnection

Reply:
What? :) Dear @gloriousCode , how you have reached this [point](https://github.com/thrasher-corp/gocryptotrader/pull/650/files#diff-81b63f155e3eb371602734d39949da22f7752344a95cf380308c9376517ca1cbR470)? How you set w.features.FullPayloadSubscribe to true if there is no place in kraken code where FullPayloadSubscribe is set.
Even if you set it in config, it is impossible until [this](https://github.com/thrasher-corp/gocryptotrader/blob/6a28bd2241ecca18f4dc696ab17b212af4e313df/exchanges/kraken/kraken_wrapper.go#L128)  is true.

Reply:
Even if I manually set 

WebsocketCapabilities: protocol.Features{
  FullPayloadSubscribe: true,
  Subscribe:          false,
 ...

I have no problem with reconnecting.

Reply:
My mistake :D [this line](https://github.com/thrasher-corp/gocryptotrader/pull/650/files#diff-81b63f155e3eb371602734d39949da22f7752344a95cf380308c9376517ca1cbL434)

I don't have problems reconnecting, but after a disconnection and reconnection, I never re-subscribe

Reply:
Dear Scott, I have found one more important pidarastic bug, which affected on our issue with Kraken! (and may be not only Kraken) :)
So please check if Kraken reconnects and re-subscribes successfully in your case. Thanks!

Reply:
While I appreciate the lock fix, I don't believe that it is what is stopping resubscription like I'm saying. Would you be able to send me your verbose output for Kraken websocket, because here is mine on your branch (with irrelevant data removed):
```
[WARN] |  29/03/2021 09:47:33  | Internet connectivity lost
[ERROR] |  29/03/2021 09:47:33  | Kraken Error connecting Kraken websocket connection: wss://ws.kraken.com Error: dial tcp: lookup ws.kraken.com: no such host
[DEBUG] |  29/03/2021 09:47:33  | Kraken websocket: running connection monitor cycle
[DEBUG] |  29/03/2021 09:47:36  | Internet connectivity re-established
[DEBUG] |  29/03/2021 09:47:36  | Order manager: Processing orders for exchange Kraken.
[INFO] |  29/03/2021 09:47:37  | Kraken Websocket connected to wss://ws.kraken.com
[DEBUG] |  29/03/2021 09:47:37  | Kraken websocket: running connection monitor cycle
[DEBUG] |  29/03/2021 09:47:37  | Kraken websocket connection: message received: {"connectionID":11375951781627363295,"event":"systemStatus","status":"online","version":"1.8.2"}
[WARN] |  29/03/2021 09:47:37  | Kraken New version of Websocket API released. Was 1.4.0 Now 1.8.2
[DEBUG] |  29/03/2021 09:47:39  | Kraken websocket: running connection monitor cycle
[DEBUG] |  29/03/2021 09:47:41  | Kraken websocket: running connection monitor cycle
[DEBUG] |  29/03/2021 09:47:44  | Kraken websocket: running connection monitor cycle
[DEBUG] |  29/03/2021 09:47:46  | Kraken websocket: running connection monitor cycle
[DEBUG] |  29/03/2021 09:47:46  | Order manager: Processing orders for exchange Kraken.
[DEBUG] |  29/03/2021 09:47:50  | Kraken websocket: running connection monitor cycle
[DEBUG] |  29/03/2021 09:47:52  | Kraken websocket: running connection monitor cycle
[DEBUG] |  29/03/2021 09:47:54  | Kraken websocket: running connection monitor cycle
[DEBUG] |  29/03/2021 09:47:54  | Kraken websocket connection: sending message [{"event":"ping"}]
[DEBUG] |  29/03/2021 09:47:54  | Kraken websocket connection: message received: {"event":"pong"}
```

and here is the output on `master`
```
[WARN] |  29/03/2021 09:51:50  | Internet connectivity lost
[ERROR] |  29/03/2021 09:51:52  | Kraken Error connecting Kraken websocket connection: wss://ws.kraken.com Error: dial tcp: lookup ws.kraken.com: no such host
[DEBUG] |  29/03/2021 09:51:52  | Kraken websocket: running connection monitor cycle
[WARN] |  29/03/2021 09:51:53  | Order manager: Unable to get active orders for Kraken and asset type spot: Post "https://api.kraken.com/0/private/OpenOrders": dial tcp: lookup api.kraken.com: no such host
[DEBUG] |  29/03/2021 09:51:54  | Internet connectivity re-established
[INFO] |  29/03/2021 09:51:55  | Kraken Websocket connected to wss://ws.kraken.com
[DEBUG] |  29/03/2021 09:51:55  | Kraken websocket connection: message received: {"connectionID":8358636366362181212,"event":"systemStatus","status":"online","version":"1.8.2"}
[WARN] |  29/03/2021 09:51:55  | Kraken New version of Websocket API released. Was 1.4.0 Now 1.8.2
[DEBUG] |  29/03/2021 09:51:56  | Kraken websocket connection: sending message to websocket {Event:subscribe RequestID:195851083 Pairs:[XBT/USD] Subscription:{Name:book Interval:0 Depth:1000 Token:} Channels:[{Channel:book Currency:XBT/USD Asset:spot Params:map[]}]}
[DEBUG] |  29/03/2021 09:51:56  | Kraken websocket connection: message received: {"channelID":320,"channelName":"book-1000","event":"subscriptionStatus","pair":"XBT/USD","reqid":195851083,"status":"subscribed","subscription":{"depth":1000,"name":"book"}}
[DEBUG] |  29/03/2021 09:51:56  | Kraken websocket connection: sending message to websocket {Event:subscribe RequestID:126496604 Pairs:[XBT/USD] Subscription:{Name:ohlc Interval:0 Depth:0 Token:} Channels:[{Channel:ohlc Currency:XBT/USD Asset:spot Params:map[]}]}
[DEBUG] |  29/03/2021 09:51:56  | Kraken websocket connection: message received: {"channelID":327,"channelName":"ohlc-1","event":"subscriptionStatus","pair":"XBT/USD","reqid":126496604,"status":"subscribed","subscription":{"interval":1,"name":"ohlc"}}
[DEBUG] |  29/03/2021 09:51:56  | Kraken websocket connection: message received: [327,["1616971889.049037","1616971920.000000","55426.90000","55426.90000","55426.90000","55426.90000","55426.90000","0.00990000",1],"ohlc-1","XBT/USD"]
[DEBUG] |  29/03/2021 09:51:56  | Kraken websocket connection: sending message to websocket {Event:subscribe RequestID:133299090 Pairs:[XBT/USD] Subscription:{Name:spread Interval:0 Depth:0 Token:} Channels:[{Channel:spread Currency:XBT/USD Asset:spot Params:map[]}]}
[DEBUG] |  29/03/2021 09:51:57  | Kraken websocket connection: message received: [320,{"as":[["55426.90000","0.00998266","1616971889.049037"],["55452.70000","0.09565900","1616971892.240863"],["55465.50000","0.11495177","1616971913.885107"],["55465.60000","0.78841103","1
```